src/unit_tests/wazuh_modules_scheduling/test_schedule_scan.c

```diff
@@ -0,0 +1,441 @@
+/**
+ * Unitary test for methods 
+ * described in 'headers/schedule_scan.h' and 
+ * 'shared/schedule_scan.c' files
+* */
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include <time.h> 
+#include "shared.h"
+#include "wazuh_modules/wmodules.h"
+
+static time_t current_time = 0;
+
+extern time_t _get_next_time(const sched_scan_config *config, const char *MODULE_TAG,  const int run_on_start);
+extern int _sched_scan_validate_parameters(sched_scan_config *scan_config);
+extern time_t __real_time(time_t *_time);
+
+typedef struct test_structure {
+    xml_node **nodes;
+    sched_scan_config *scan_config;
+} test_structure;
+/*********************************/
+/*       WRAPS                   */ 
+/*********************************/
+int __wrap_OS_StrIsNum(const char *str) {
+    int retval = mock();
+
+    check_expected(str);
+
+    return retval;
+}
+
+void __wrap__merror(const char * file, int line, const char * func, const char *msg, ...)
+{
+    char formatted_msg[OS_MAXSTR];
+    va_list args;
+
+    va_start(args, msg);
+    vsnprintf(formatted_msg, OS_MAXSTR, msg, args);
+    va_end(args);
+
+    check_expected(formatted_msg);
+}
+
+void __wrap__mwarn(const char * file, int line, const char * func, const char *msg, ...)
+{
+    return;
+}
+
+int __wrap_check_day_to_scan(int day, const char *hour) {
+    check_expected(day);
+    check_expected(hour);
+    return mock();
+}
+
+int __wrap_get_time_to_hour(const char * hour) {
+    check_expected(hour);
+    return mock();
+}
+
+int __wrap_get_time_to_day(int wday, const char * hour) {
+    check_expected(wday);
+    check_expected(hour);
+    return mock();
+}
+
+time_t __wrap_time(time_t *_time){
+    if(!current_time){
+        current_time = __real_time(NULL);
+    }
+    return current_time;
+}
+
+void __wrap_wm_delay(unsigned int msec){
+    current_time += (msec/1000);
+}
+/*********************************/
+/*       SETUP-TEARDOWN          */
+/*********************************/
+static int test_scan_read_setup(void **state) {
+    test_structure *test; 
+    os_calloc(1, sizeof(test_structure), test);
```

Comment:
Please change calls to `os_calloc` with calls to `calloc` to ensure merror_exit is not called if there is an error, causing an early finish of the tests.
In order to keep consistency I would also change calls to `os_free` with calls to `free`.